pkg/netpol/eval/exposure.go

```diff
@@ -147,3 +95,97 @@ func (pe *PolicyEngine) GetPeerNsLabels(p Peer) (map[string]string, error) {
 	}
 	return peer.PotentialNamespaceLabels, nil
 }
+
+// IsRedundantRepresentativePeer returns wether the given peer is a redundant representative peer or not
+func (pe *PolicyEngine) IsRedundantRepresentativePeer(peer Peer) bool {
+	if !pe.IsRepresentativePeer(peer) {
+		return false
+	}
+	return peer.(*k8s.RepresentativePeer).IsRedundant
+}
+
+/////////
+
+// RepresentativePeerMatchesRealWorkloadPeer gets two peers
+// if one is a WorkloadPeer and the other is a RepresentativePeer: returns whether the namespaceLabels (TODO: and podLabels)
+// of the RepresentativePeer are contained in the labels of the WorkloadPeer's namespace
+func (pe *PolicyEngine) RepresentativePeerMatchesRealWorkloadPeer(firstPeer, secondPeer Peer) (bool, error) {
+	var repPeer *k8s.RepresentativePeer
+	var wlPeer *k8s.WorkloadPeer
+	if pe.IsRepresentativePeer(firstPeer) {
+		repPeer = firstPeer.(*k8s.RepresentativePeer)
+		// if the second is not a workload peer - no match for sure ; return false
+		if wlPeer, _ = isPeerAWorkloadPeer(secondPeer); wlPeer == nil {
+			return false, nil
+		}
+		// TODO : compare also podLabels
+		return pe.potentialNamespaceLabelsContainedInRealLabels(repPeer, wlPeer)
+	} // else
+	wlPeer, _ = isPeerAWorkloadPeer(firstPeer)
+	if !pe.IsRepresentativePeer(secondPeer) || wlPeer == nil {
+		return false, nil
+	}
+	repPeer = secondPeer.(*k8s.RepresentativePeer)
+	return pe.potentialNamespaceLabelsContainedInRealLabels(repPeer, wlPeer)
+}
+
+// potentialNamespaceLabelsContainedInRealLabels checks if the potential namespaceLabels of the fake peer's namespace are contained in
+// the labels of the namespace of the real workloadpeer;
+// if yes, mark the representative peer as redundant
+func (pe *PolicyEngine) potentialNamespaceLabelsContainedInRealLabels(repPeer *k8s.RepresentativePeer,
+	wlPeer *k8s.WorkloadPeer) (bool, error) {
+	potentialNsLabels := repPeer.PotentialNamespaceLabels
+	realPodPeer, err := pe.convertPeerToPodPeer(wlPeer)
+	if err != nil {
+		return false, err
+	}
+	actualWlNsLabels := realPodPeer.NamespaceObject.Labels
+	// check if the potential labels map contains a name label key - compare with the actual ns name and ignore it
+	// while compare labels maps
+	nameLabelKey := checkForNameLabelsInTheMap(potentialNsLabels)
+	// if the name value is different from the realPod namespace name then no match
+	if nameLabelKey != "" && realPodPeer.NamespaceObject.Name != potentialNsLabels[nameLabelKey] {
+		return false, nil
+	}
+	isRedundant := isMapContainedInOther(potentialNsLabels, actualWlNsLabels, nameLabelKey)
+	repPeer.IsRedundant = isRedundant
+	return isRedundant, nil
+}
+
+// isMapContainedInOther checks if the subMap is contained in the other map;
+// ignored key : is the name key; its value is already compared so to be ignored if not empty
```

Comment:
why is this key ignored? 

+ cannot use existing methods for checking selectors matching ?

Reply:
- the name is compared earlier with the actual name of the namespace. 
so skipped (ignored) checking if it is contained in the real labels. 
(i.e. if a representative peer has selector `name: hello-world`; and we have real namespace  with `hello-world` name; we might not see this label in the real namespace; but we want to compare other labels to see if should remove the representative peer or not)
- at this point I had the selectors as maps not Selector types. (so couldn't use checking selectors matching funcs)

Reply:
maybe it is possible to convert maps to selector types? or create selectors from maps?

Reply:
both are acceptable;
for example: 
if we have a rule in the policy : 
```
from:
    - namespaceSelector:       
       matchLabels:
          kubernetes.io/metadata.name: payments
```

it will match any namespace with this name (even if the namespace dosen't has a label in its labels list with same  `kubernetes.io/metadata.name: payments`.

Reply:
so while comparing potentialNamespceLabels of representativePeer and a real workload
I first check if the representative peer has this "name" label in its `potentialNamespceLabels ` ; if yes I comapre the value of the name key with the real workload's namespace name:
- if they are different then I return that there is no match between the peers
- if they are the same; I continue to compare the other labels in `potentialNamespceLabels ` are contained in the real namespace labels

Reply:
but we add the label `kubernetes.io/metadata.name` to any namespace that we analyze...

Reply:
oops, forgot about that

Reply:
this func was deleted - handling comparison with real peers code was changed